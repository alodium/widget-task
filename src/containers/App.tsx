import * as React from 'react';

export class App extends React.Component {
    
    render(){
        return(
            <div className="container">
                <div>i present.. the widget</div>
            </div>
        );
    }


}

export default App;