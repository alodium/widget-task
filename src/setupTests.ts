import * as enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';

// https://stackoverflow.com/questions/46435558/could-not-find-declaration-file-for-enzyme-adapter-react-16
(enzyme as any).configure({ adapter: new Adapter() });