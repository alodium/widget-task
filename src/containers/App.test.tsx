import * as React from 'react';
import { shallow }from 'enzyme';

import App from './App';
import '../setupTests';

describe('App', () => {
    const app = shallow(<App />);

    it('renders App', () => {
        expect(app).toMatchSnapshot();
    });
});