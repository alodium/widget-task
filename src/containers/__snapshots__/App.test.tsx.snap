// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        i present.. the widget
</div>,
      "className": "container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "i present.. the widget",
      },
      "ref": null,
      "rendered": "i present.. the widget",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          i present.. the widget
</div>,
        "className": "container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "i present.. the widget",
        },
        "ref": null,
        "rendered": "i present.. the widget",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
